Library/Homebrew/requirements/macos_requirement.rb

```diff
@@ -14,8 +14,10 @@ def initialize(tags = [], comparator: ">=")
     begin
       @version = if comparator == "==" && tags.first.respond_to?(:map)
         tags.shift.map { |s| MacOS::Version.from_symbol(s) }
+      elsif tags.first.is_a?(Symbol)
+        MacOS::Version.from_symbol(tags.shift)
       else
-        MacOS::Version.from_symbol(tags.shift) unless tags.empty?
+        MacOS::Version.new(tags.shift) unless tags.empty?
```

Comment:
Why do we need to support the non-symbol version? Wouldn't it be nicer to stick to a single format?

Reply:
It was supported for casks before. We can of course merge https://github.com/Homebrew/homebrew-cask/pull/90326 and remove support for this completely instead.

Reply:
> We can of course merge [Homebrew/homebrew-cask#90326](https://github.com/Homebrew/homebrew-cask/pull/90326) and remove support for this completely instead.

I'd be 👍🏻 on that.